### PR TITLE
Some modifications in mwoauth.request method

### PR DESCRIPTION
**Pull request by urbanecm**

### CODE REVIEW

**Comment by urbanecm** [comment]
@valhallasw Can you please review this? I'd like to use this library, but such things kinda block me.

**Comment by valhallasw** [comment]
Aah! I'm so sorry - I thought I had reviewed it, but apparently I had not clicked the 'submit review' button. I'm not sure if the comments are still completely valid (they are marked as outdated by github). Please take a look at them, and I'll reserve some time to re-review the patches in the coming week.

File: flask_mwoauth/__init__.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def request(self, api_query, url=None):
                 client_secret=self.consumer_token.secret,
                 resource_owner_key=session['mwoauth_access_token']['key'],
                 resource_owner_secret=session['mwoauth_access_token']['secret'])
-            return requests.post(api_url, data=api_query, auth=auth1).text
+            return requests.post(api_url, data=api_query, auth=auth1).json()
```
[the remark this answers is not in the document]
Comment (urbanecm) [reply flask_mwoauth/__init__.py]: Thank you for notifing. Fixed.

**Comment by urbanecm** [comment]
Thank you for reviewing! I have fixed one raised issue and commented on the other one.

File: flask_mwoauth/__init__.py [hunk 2]
```diff
@@ -142,7 +142,12 @@ def request(self, api_query, url=None):
             req.send()
             return req.response.text
         else:
-            return requests.post(api_url, data=api_query).text
+            auth1 = OAuth1(
```
[the remark this answers is not in the document]
Comment (valhallasw) [reply flask_mwoauth/__init__.py]: I think that would be a bit cleaner, but I think it's also OK to leave this until later. A little bit of duplication close together isn't the end of the world :-)

**Comment by valhallasw** [comment]
With the backwards compatibility issue resolved I'll merge the request. I have to figure out again how to do a release on pypi, but will do that somewhere in the coming week. (if I forget, please don't hesitate to remind me)

**Comment by valhallasw** [comment]
Whoops, I just missed that there's also the `             return req.response.json()` on line 145. I'll fix that before uploading to pypi.

**Comment by urbanecm** [comment]
ahh, sorry, didn't notice it.

**Comment by urbanecm** [comment]
Created #22

**Comment by valhallasw** [comment]
Released as https://pypi.org/project/flask-mwoauth/0.3.70/, thanks for your help!

**Pull request merged**